kyuubi-zookeeper/src/main/scala/org/apache/kyuubi/zookeeper/EmbeddedZookeeper.scala

```diff
@@ -31,12 +31,18 @@ class EmbeddedZookeeper extends AbstractService("EmbeddedZookeeper") {
   private var zks: ZooKeeperServer = _
   private var serverFactory: NIOServerCnxnFactory = _
   private var dataDirectory: File = _
+  private var dataLogDirectory: File = _
   // TODO: Is it right in prod?
   private val deleteDataDirectoryOnClose = true
   private var host: String = _
 
   override def initialize(conf: KyuubiConf): Unit = synchronized {
-    dataDirectory = new File(conf.get(ZK_DATA_DIR))
+    val kyuubiHome = sys.env.getOrElse(KyuubiConf.KYUUBI_HOME, ".")
+    val kyuubiHomeWithDollar = "$" + KyuubiConf.KYUUBI_HOME
```

Comment:
the idea looks cool, but what if the user configures it to `${KYUUBI_HOME}/embedded_zookeeper` or `$ANOTHER_ENV_VAR/embedded_zookeeper`?

Reply:
Also, as there is a behavior change, we'd better leave a notification at migration guide

Reply:
> the idea looks cool, but what if the user configures it to `${KYUUBI_HOME}/embedded_zookeeper` or `$ANOTHER_ENV_VAR/embedded_zookeeper`?



> the idea looks cool, but what if the user configures it to `${KYUUBI_HOME}/embedded_zookeeper` or `$ANOTHER_ENV_VAR/embedded_zookeeper`?

In my opinion: 
If we allow  configuration with system env and java properties, it should be implemented in `KyuubiConf` class, `KyuubiConf` can substitute the variables when load all the values. 
 Generating zk directories everywhere annoys me.  This PR is just to ensure that it works well out of the box. 
Or we should set the default dir under `java.io.tmpdir`?

Reply:
> Also, as there is a behavior change, we'd better leave a notification at migration guide

I see, but which version (from/to) should I use？

Reply:
@biangjuang for simple, what do you think about just changing the semantic of the relative path, from `$PWD` to the `$KYUUBI_HOME`? Then we don't need to handle the environment variable resolution (such interpret behavior may involve security issues), just update the migration guide to mention it (I plan to cut the 1.8.0 RC at late this week, if the PR complete before Tuesday, including 1.8 is fine)